genesyscloud/telephony_providers_edges_phone/resource_genesyscloud_telephony_providers_edges_phone.go

```diff
@@ -0,0 +1,273 @@
+		return diagErr
+	}
+
+	return readPhone(ctx, d, meta)
+}
+
+func readPhone(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	pp := getPhoneProxy(sdkConfig)
+
+	log.Printf("Reading phone %s", d.Id())
+	return gcloud.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		currentPhone, resp, getErr := pp.getPhoneById(ctx, d.Id())
+		if getErr != nil {
+			if gcloud.IsStatus404(resp) {
+				return retry.RetryableError(fmt.Errorf("failed to read phone %s: %s", d.Id(), getErr))
+			}
+			return retry.NonRetryableError(fmt.Errorf("failed to read phone %s: %s", d.Id(), getErr))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourcePhone())
+		d.Set("name", *currentPhone.Name)
+		d.Set("state", *currentPhone.State)
+		d.Set("site_id", *currentPhone.Site.Id)
+		d.Set("phone_base_settings_id", *currentPhone.PhoneBaseSettings.Id)
+		d.Set("line_base_settings_id", *currentPhone.LineBaseSettings.Id)
+		if currentPhone.PhoneMetaBase != nil {
+			d.Set("phone_meta_base_id", *currentPhone.PhoneMetaBase.Id)
+		}
+
+		if currentPhone.WebRtcUser != nil {
+			d.Set("web_rtc_user_id", *currentPhone.WebRtcUser.Id)
+		}
+
+		if currentPhone.Lines != nil {
+			d.Set("line_addresses", flattenPhoneLines(currentPhone.Lines))
+		}
+
+		if currentPhone.Capabilities != nil {
+			d.Set("capabilities", flattenPhoneCapabilities(currentPhone.Capabilities))
+		}
+
+		log.Printf("Read phone %s %s", d.Id(), *currentPhone.Name)
+		return cc.CheckState()
+	})
+}
+
+func updatePhone(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	pp := getPhoneProxy(sdkConfig)
+
+	name := d.Get("name").(string)
+	site := gcloud.BuildSdkDomainEntityRef(d, "site_id")
+	phoneBaseSettings := buildSdkPhoneBaseSettings(d, "phone_base_settings_id")
+	phoneMetaBase := gcloud.BuildSdkDomainEntityRef(d, "phone_meta_base_id")
+	webRtcUserId := d.Get("web_rtc_user_id")
+
+	var err error
+	lineBaseSettingsID := d.Get("line_base_settings_id").(string)
+	if lineBaseSettingsID == "" {
+		lineBaseSettingsID, err = getLineBaseSettingsID(ctx, pp, *phoneBaseSettings.Id)
+		if err != nil {
+			return diag.Errorf("Failed to get line base settings for %s: %s", name, err)
+		}
+	}
+
+	lineBaseSettings := &platformclientv2.Domainentityref{Id: &lineBaseSettingsID}
+	lines, isStandalone := buildSdkLines(ctx, pp, d, lineBaseSettings)
+
+	//Have to create a phonebasesettings object now as of version v90.  Used to be a domain ref but the engineering team changed the type in the swagger def
+	phoneSettings := &platformclientv2.Phonebasesettings{
+		Id: phoneBaseSettings.Id,
+	}
+
+	updatePhoneBody := &platformclientv2.Phone{
+		Name:              &name,
+		Site:              site,
+		PhoneBaseSettings: phoneSettings,
+		PhoneMetaBase:     phoneMetaBase,
+		LineBaseSettings:  lineBaseSettings,
+		Lines:             lines,
+	}
```

Comment:
Same idea as with the create function, you can set some of the properties on the object directly instead of using variables. It might also help to create a `getPhoneFromResourceData` function like in the other packages to help reduce repetitive code in the create and update functions

Reply:
Created method `getPhoneFromResourceData` for use by create and update